cache/ringbuffer.go

```diff
@@ -73,29 +75,117 @@ func (r *ringBuffer) append(b batch) {
 	r.head = (r.head + 1) % len(r.buffer)
 }
 
-// Filter returns all events in the buffer whose ModRevision is >= minRev.
+// AscendGreaterOrEqual iterates through batches in ascending order starting from the first batch with rev >= pivot.
 // TODO: use binary search on the ring buffer to locate the first entry >= nextRev instead of a full scan
-func (r *ringBuffer) Filter(minRev int64) (eventBatches [][]*clientv3.Event) {
-	eventBatches = make([][]*clientv3.Event, 0, r.size)
+func (r *ringBuffer) AscendGreaterOrEqual(pivot int64, iter IterFunc) {
+	if r.size == 0 {
+		return
+	}
+
+	for logicalIdx := 0; logicalIdx < r.size; logicalIdx++ {
```

Comment:
What's wrong with `for n, i := 0, r.tail; n < r.size; n, i = n+1, (i+1)%len(r.buffer) {`. For me it's more readable to just work on physical index.